apps/easypid/src/agent/index.ts

```diff
@@ -1,10 +1,38 @@
-import { type EasyPIDAppAgent, useAgent } from '@package/agent'
+import { useAgent } from '@package/agent'
+import { type EasyPIDAppAgent, type ParadymAppAgent, isEasyPIDAgent, isParadymAgent } from '@package/agent/src/agent'
 import { useSecureUnlock as _useSecureUnlock } from '@package/secure-store/secureUnlock'
+import type { initializeAppAgent } from './initialize'
 
 export { initializeAppAgent } from './initialize'
 
-export const useAppAgent = useAgent<EasyPIDAppAgent>
-export type AppAgent = EasyPIDAppAgent
+export type AppAgent = Awaited<ReturnType<typeof initializeAppAgent>>
```

Comment:
What is the use of this second agent type?

Reply:
Okay, but since we have only one app now, this will not really work right? Since we only know this during runtime, and not beforehand.

To fix this, I now made a second agent hook that is used for didcomm specific features, so you know you are using an agent that actually has didcomm enabled (see below).

Reply:
Yes that makes sense. It should probably be the type of `EitherAgent` now (so you can only use feature available in both). Otherwise you need a specific hook. 

Could be e.g. `useDidcommAgent` even so it's more generic, but just `useParadymAgetn` and `useEasyPidAgent` are fine as well (it's temporary anyway I guess)